src/raster/gcp.rs

```diff
@@ -20,11 +123,83 @@ impl Dataset {
 
         unsafe { SpatialRef::from_c_obj(c_ptr) }.ok()
     }
+
+    /// Fetch GCPs.
+    ///
+    /// See: [`GDALDataset::GetGCPs`](https://gdal.org/api/gdaldataset_cpp.html#_CPPv4N11GDALDataset7GetGCPsEv)
+    pub fn gcps(&self) -> &[GcpRef] {
+        let len = unsafe { gdal_sys::GDALGetGCPCount(self.c_dataset()) };
+        let data = unsafe { gdal_sys::GDALGetGCPs(self.c_dataset()) };
+        unsafe { std::slice::from_raw_parts(data as *const GcpRef, len as usize) }
+    }
```

Comment:
What's the argument against returning `Vec<Gcp>`, which (I think) would eliminate the need for `GcpRef`. Because of the data copies that would result? 

I partly ask because going from Rust to C we do a data copy (from `Gcp` to `GDAL_GCP`), but from C to Rust we have an extra `transparent`, castable structure `GcpRef` to avoid that copy, which seems asymmetric.

Reply:
2N+1 allocations. That's what I originally had.

It's asymmetric because of the string conversions. We can do C to Rust string conversions on request (in the getter), but we can't to do it the other way around because C will always read the strings.